### DISCUSSION TITLE
How do I tell Renovate which Python version I'm using with requirements.txt?

### DISCUSSION THREAD

**Discussion by embeddedt** [renovatebot/renovate#10538] [Request Help]
Hi all,

I'm working on setting up automatic dependency upgrades for a private project, and ran into a bit of an issue.

`djangorestframework-simplejwt` is one of my dependencies, and I'm currently using 4.4.0. Renovate is trying to automatically upgrade it to 4.7.1 as that is the latest version on PyPI. However, the newer version only supports Python 3.7, and I'm still using Python 3.6. Thus, my CI fails on the pull request with the following error:

```
ERROR: Could not find a version that satisfies the requirement djangorestframework-simplejwt==4.7.1 (from versions: 1.0, 1.1, 1.2, 1.2.1, 1.3, 1.4, 1.5.1, 2.0.0, 2.0.1, 2.0.2, 2.0.3, 2.0.4, 2.0.5, 2.1, 3.0, 3.1, 3.2, 3.2.1, 3.2.2, 3.2.3, 3.3, 4.0.0, 4.1.0, 4.1.1, 4.1.2, 4.1.3, 4.1.4, 4.1.5, 4.2.0, 4.3.0, 4.4.0)
ERROR: No matching distribution found for djangorestframework-simplejwt==4.7.1
```

Is there a way to tell Renovate that I'm using Python 3.6 and thus this upgrade is impossible?

**Comment by viceice** [comment]
Nope, we don't check transitive dependencies. You need to manually create a package rule and use `allowedVersions` to limit possible versions

https://docs.renovatebot.com/configuration-options/#allowedversions

**Reply by rarkins** [reply]
Is it transitive or not? I thought what he needs is constraints.python

**Reply by embeddedt** [reply]
@rarkins Thanks; that looks like exactly what I need.
